### DISCUSSION TITLE
Implement ZSTD as a middleware for compression

### DISCUSSION THREAD

**Discussion by Kumzy** [litestar-org/litestar#1537] [Ideas]
Implement ZStandard library to compress data.

Media type: application/zstd

On our current API (flask), we implemented ZSTD to compress huge JSON response that were over 5MB using the default compression level (3) and resulted in some ~200Kb JSON. We use it for customers that have slow networks/low bandwidth.

As it's not native/supported by browsers, I do not think it can be used to compress html.

Links:
- https://www.rfc-editor.org/rfc/rfc8478
- https://github.com/facebook/zstd

**Comment by provinzkraut** [comment]
What library are you currently using for this?

**Reply by Kumzy** [reply]
We are using https://pypi.org/project/zstandard this one but it seems that a lot of python implementation of zstd exists

**Comment by JacobCoffee** [comment]
https://github.com/litestar-org/litestar/pull/4286
